### PR TITLE
Set up http testing with vcr

### LINKED ISSUE #19: Set up HTTP testing
Set up HTTP testing with the [vcr](https://github.com/ropensci/vcr) package to allow for testing of the methods making API requests without hitting the API every time.

### PR DESCRIPTION
**Pull request by pr130**
closes #19 
This PR sets up HTTP testing using [vcr](https://github.com/ropensci/vcr) and implements some sample tests in `test-kobo-client.R` and `test-kobo.R`. 

I heavily relied on those two chapters from the HTTP testing book:
- https://books.ropensci.org/http-testing/vcr.html
- https://books.ropensci.org/http-testing/vcr-intro.html (and the subsequent pages in the "vcr details" chapter)

**How does it work?** 
1. when we wrap a test with vcr (see how to do that [here](https://books.ropensci.org/http-testing/vcr.html#ps-where-to-put-use_cassette)) and run it for the first time, vcr serializes the response from the server into a `yml` and puts it into the `tests/fixtures` folder with the filename that we specified as argument to `use_cassette`. 
2. Later on, when we re-run the test, code is run as usual (the url is built, etc) **until** the private `make_request` function of HttpClient checks that vcr is enabled -> instead of making the request to the server, the cassette (=the yml) is loaded and the request that matches the request _that was about to be made_ is replayed instead. The matching is based on the URI (e.g. "https://kobo.correlaid.org/api/v2/assets?format=json") and the HTTP method (`get`, `post` etc)
3. If you change the request in the test, make sure to delete the yml file so that vcr can record a new cassette for the changed test. 

**Gotchas**
because the matching is done based comparing the URI in the yml with the URI in the _prepared_ request, it is susceptible to changes being made by the server. 
In a very annoying case, vcr gave me an error saying it could not find a matching cassette for the assets endpoint even though i had recorded it. After [enabling the vcr logs](https://books.ropensci.org/http-testing/debugging-your-tests-that-use-vcr.html#logging), i found the following line:

```
[Cassette: 'kobo-get-404'] - 2021-05-10 15:19:45 -     uri did not match: current request [get https://kobo.correlaid.org/api/v2/assets?format=json] vs [get https://kobo.correlaid.org/api/v2/assets/?format=json]
```

the first one is the URI of the prepared request , the second one the URI recorded in the yml file. note the missing / before the ?.

Basically, whenever we make a request without a trailing `/` the kobo server will append it while processing the request, resulting in a mismatch later on when vcr tries to match the "prepared" request (without "/") with the serialized response in the yml (which has the "/" because kobo has added it). Hence, we should always pass the path with the trailing slash. I'll open a separate issue for that.

### CODE REVIEW

**Comment by pr130** [comment]
sorry for the bunch of commits - i always merged from dev ;) let's squash-merge this one. 
Also, i'll write up a small wiki post over the coming days - just wanted to put this into `dev`.

**Comment by pr130** [comment]
issue for trailing slash issue #26

**Comment by MKyhos** [comment]
Local `devtools::test()` passes, expect of  [this line](https://github.com/CorrelAid/kbtbr/blob/httptesting/tests/testthat/test-kobo.R#L38) caused some trouble: 

- instead of the expected `SSL: no alternative certificate subject name matches target host name 'nokobohere.correlaid.org'`, 
- for me the error message should be `SSL certificate problem: unable to get local issuer certificate`

**Review by MKyhos — COMMENTED** [review]
I could not see any obvious faults etc, well done! 

Edit: **LGTM** :+1:

**Comment by pr130** [comment]
thanks for the thorough review and not detecting any obvious faults etc. @MKyhos !
